hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2678,3 +2679,115 @@ def hpat_pandas_series_fillna_impl(self, value=None, method=None, axis=None, inp
                 return pandas.Series(filled_data, self._index, self._name)
 
             return hpat_pandas_series_fillna_impl
+
+
+@overload_method(SeriesType, 'pct_change')
+def hpat_pandas_series_pct_change(self, periods=1, fill_method='pad', limit=None, freq=None):
+    """
+    Pandas Series method :meth:`pandas.Series.pct_change` implementation.
+
+    Note: Unsupported mixed numeric and string data
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests -k hpat.tests.test_series.TestSeries.test_series_pct_change
+       Test: python -m hpat.runtests -k hpat.tests.test_series.TestSeries.test_series_pct_change_str
+       Test: python -m hpat.runtests -k hpat.tests.test_series.TestSeries.test_series_pct_change_not_supported
+
+    Parameters
+    -----------
+    self: :obj:`pandas.Series`
+        input series
+    periods: :obj:`int`, default 1
+        Periods to shift for forming percent change.
+    fill_method: :obj:`str`, default 'pad'
+        How to handle NAs before computing percent changes.
+    limit: :obj:`int`, default Nogne
+        The number of consecutive NAs to fill before stopping.
+        *unsupported*
+    freq: :obj: DateOffset, timedelta, or offset alias string, optional
+        Increment to use from time series API (e.g. 'M' or BDay()).
+        *unsupported*
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns :obj:`pandas.Series` object
+    """
+
+    _func_name = 'Method pct_change().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError(
+            '{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    if not isinstance(self.data.dtype, types.Number):
+        raise TypingError(
+            '{} The function only applies to elements that are all numeric. Given data type: {}'.format(_func_name, self.data.dtype))
+
+    if not isinstance(periods, (types.Integer, types.Omitted)):
+        raise TypingError(
+            '{} The function uses only periods is integer. Given periods type: {}'.format(_func_name, periods))
+
+    if not isinstance(fill_method, (str, types.UnicodeType, types.StringLiteral, types.NoneType, types.Omitted)):
+        raise TypingError(
+            '{} The function uses only fill_method is string. Given fill_method type: {}'.format(_func_name, fill_method))
+
+    if not isinstance(limit, (types.Omitted, types.NoneType)):
+        raise TypingError(
+            '{} The function unsupport limit is not None.'.format(_func_name))
+
+    if not isinstance(freq, (types.Omitted, types.NoneType)):
+        raise TypingError(
+            '{} The function unsupport freq is not None.'.format(_func_name))
```

Comment:
Please use type checker to validate parameters types. You can find example of usage here:
https://github.com/densmirn/hpat/blob/5d99d7dd1500df112651937ce2d925dada4c6c2b/hpat/datatypes/hpat_pandas_series_functions.py#L1330-L1341.